cmd/thanos/main.go

```diff
@@ -212,6 +212,38 @@ func registerMetrics(mux *http.ServeMux, g prometheus.Gatherer) {
 	mux.Handle("/metrics", promhttp.HandlerFor(g, promhttp.HandlerOpts{}))
 }
 
+func registerHealthy(mux *http.ServeMux, logger log.Logger, component string) {
+	mux.HandleFunc("/-/healthy", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := fmt.Fprintf(w, "Thanos %v is Healthy.\n", component); err != nil {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			level.Error(logger).Log("msg", "Could not write health check response.")
+		}
+	})
+}
+
+func registerReady(mux *http.ServeMux, logger log.Logger, component string, readinessFunc func() (bool, error)) {
+	mux.HandleFunc("/-/ready", func(w http.ResponseWriter, r *http.Request) {
+		ready, err := readinessFunc()
```

Comment:
Can we do similar as here? https://github.com/iksaif/pushgateway/blob/9bd7cbfccc9577cdc120503f105b3d2508e58acc/handler/misc.go

It will simplifiy logic a lot. Just have `readinessFunc` return error. If it's nil it's ok. If it is not nil then not ready. It is clear contract (: What do you think?